api/src/stampy_chat/db/session.py

```diff
@@ -0,0 +1,82 @@
+import time
+import logging
+from contextlib import contextmanager
+from sqlalchemy import create_engine
+from sqlalchemy.orm import Session
+from sqlalchemy.exc import SQLAlchemyError
+from stampy_chat.env import DB_CONNECTION_URI
+
+
+logger = logging.getLogger(__name__)
+
+# We create a single engine for the entire application
+engine = create_engine(DB_CONNECTION_URI, echo=False)
+
+
+@contextmanager
+def make_session(auto_commit=False):
+    with Session(engine, autoflush=False) as session:
+        yield session
+        if auto_commit:
+            session.commit()
+
+
+class ItemAdder:
+    """A helper class to manage adding and flushing items to the database.
```

Comment:
this isn't needed here, but I couldn't be bothered to mess about with deleting it and inserting it elsewhere